### PR TITLE
Feature/note button on cards

### PR DESCRIPTION
**Pull request by SNujiy**

### CODE REVIEW

File: src/components/TopicMapItem/TopicMapItem.tsx [hunk 1]
```diff
@@ -23,6 +25,21 @@ export const TopicMapItem: FC<TopicMapItemProps> = ({ item, onClick }) => {
     () => [styles.topicMapItem, sizeClassname[appWidth]].join(" "),
     [appWidth],
   );
+  const [userData, setUserData] = useLocalStorage(item.id);
+  const test = localStorage;
```
Comment (henriettemoe) [inline src/components/TopicMapItem/TopicMapItem.tsx:29]: Is setUserData being used?

Or test?

Comment (SNujiy) [reply src/components/TopicMapItem/TopicMapItem.tsx:29]: No, it was removed but somehow came back after resolving merge conflicts. Will be removed again :)

File: src/components/TopicMapItem/TopicMapItem.tsx [hunk 2]
```diff
@@ -23,6 +25,20 @@ export const TopicMapItem: FC<TopicMapItemProps> = ({ item, onClick }) => {
     () => [styles.topicMapItem, sizeClassname[appWidth]].join(" "),
     [appWidth],
   );
+  const [userData] = useLocalStorage(item.id);
+  let btnState = 2;
```
Comment (boyum) [inline src/components/TopicMapItem/TopicMapItem.tsx:29]: Would it make sense to type `btnState`  with the `NoteButtonIconState` enum?

**Review by boyum — APPROVED** [review]

File: src/components/TopicMapItem/TopicMapItem.tsx (hunk 2 above)
Comment (SNujiy) [reply src/components/TopicMapItem/TopicMapItem.tsx]: Very good suggestion! Added

File: src/components/TopicMapItem/TopicMapItem.module.scss [hunk 3]
```diff
@@ -23,6 +30,10 @@
 
     .inner {
       padding: 0.75em 1em;
+
+      &.withNote:not(.noImage) {
+        padding: calc(0.75tem + $--note-btn-padding) 1em 0.75em;
```
Comment (henriettemoe) [inline src/components/TopicMapItem/TopicMapItem.module.scss:35]: Remove "t" here

```suggestion
        padding: calc(0.75em + $--note-btn-padding) 1em 0.75em;
```

File: src/components/TopicMapItem/TopicMapItem.module.scss [hunk 4]
```diff
@@ -31,6 +42,10 @@
 
     .inner {
       padding: 0.75em;
+
+      &.withNote:not(.noImage) {
+        padding: calc(0.75tem + $--note-btn-padding) 0.75em 0.75em;
```
Comment (henriettemoe) [inline src/components/TopicMapItem/TopicMapItem.module.scss:47]: Remove "t" here 

```suggestion
        padding: calc(0.75em + $--note-btn-padding) 0.75em 0.75em;
```

File: src/components/TopicMapItem/TopicMapItem.tsx [hunk 5]
```diff
@@ -41,6 +59,20 @@ export const TopicMapItem: FC<TopicMapItemProps> = ({ item, onClick }) => {
           item.topicImage?.path ? "" : styles.noImage
         }`}
```
Comment (henriettemoe) [inline src/components/TopicMapItem/TopicMapItem.tsx:60]: Add class withNote if item has dialog

```suggestion
        } ${item.dialog?.hasNote ? styles.withNote : ""}`}
```

File: src/components/TopicMapItem/TopicMapItem.module.scss [hunk 6]
```diff
@@ -1,3 +1,6 @@
+$--half-note-btn-size: 1rem;
+$--note-btn-padding: 0.75rem;
```
Comment (boyum) [inline src/components/TopicMapItem/TopicMapItem.module.scss:2]: SCSS variables do not need to be prefixed `--`, and it's against standard practice.

File: src/components/TopicMapItem/TopicMapItem.module.scss [hunk 7]
```diff
@@ -15,6 +18,10 @@
 
     .inner {
       padding: 1em 1.25em;
+
+      &.withNote:not(.noImage) {
+        padding: calc(1em + $--note-btn-padding) 1.25em 1em;
```
Comment (boyum) [inline src/components/TopicMapItem/TopicMapItem.module.scss:23]: Is the run time `calc` needed here? `$--note-btn-padding` is static and it's probably enough to do the calculation build time 
```suggestion
        padding: 1em + $--note-btn-padding 1.25em 1em;
```

File: src/components/TopicMapItem/TopicMapItem.module.scss (hunk 6 above)
Comment (henriettemoe) [reply src/components/TopicMapItem/TopicMapItem.module.scss:2]: My fault 🥲  But yes, the -- can be removed

File: src/components/TopicMapItem/TopicMapItem.module.scss (hunk 3 above)
Comment (SNujiy) [reply src/components/TopicMapItem/TopicMapItem.module.scss:35]: typo, fixed

File: src/components/TopicMapItem/TopicMapItem.module.scss (hunk 4 above)
Comment (SNujiy) [reply src/components/TopicMapItem/TopicMapItem.module.scss:47]: typo, fixed

File: src/components/TopicMapItem/TopicMapItem.tsx (hunk 5 above)
Comment (SNujiy) [reply src/components/TopicMapItem/TopicMapItem.tsx:60]: done

File: src/components/TopicMapItem/TopicMapItem.module.scss (hunk 6 above)
Comment (SNujiy) [reply src/components/TopicMapItem/TopicMapItem.module.scss]: Done

**Review by henriettemoe — APPROVED** [review]

**Pull request merged**